setup.py

```diff
@@ -566,27 +566,33 @@ def run(self):
 library_dirs.append(lib_path)
 
 # we specify exact lib names to avoid conflict with lua-torch installs
-ATEN_LIB = os.path.join(lib_path, 'libATen.so')
+ATEN_LIBS = [os.path.join(lib_path, 'libATen_cpu.so')]
+if WITH_CUDA:
+    ATEN_LIBS.extend(['-Wl,--no-as-needed', os.path.join(lib_path, 'libATen_cuda.so'), '-Wl,--as-needed'])
```

Comment:
This won't work on mac. Not all linkers support `--no-as-needed`

Reply:
But it's OK because this line is Linux only anyway (see below for the dylib path)